### PR TITLE
add some missing licenses

### PR DESCRIPTION
**Pull request by jonapich**
`MPL-2.0` from `pathspec@0.12.1`

Got this from [here](https://github.com/coveo/stew/actions/runs/8558567706/job/23453402164?pr=44) when trying to bump my dependencies.

### CODE REVIEW

**Review by dblanchette — DISMISSED** [review]

File: public.yml
Comment (JPLachance) [inline public.yml:34]: Hello @Mblanchetdeverly,

Is 'MPL-2.0' approved for public/distributed projects?

Comment (Mblanchetdeverly) [reply public.yml:34]: No its not as it is a copyleft license. It could be approved for a specific use case, if the binaries are distributed but not the code, since the obligations only apply when using code, not when linking binaries. However, we cannot approve it at large.

Comment (jonapich) [reply public.yml:33]: Hi @Mblanchetdeverly, in this case:

- This is introduced by the `pathspec` module ([link](https://github.com/cpburnz/python-pathspec/tree/master?tab=readme-ov-file))
- It is only used for testing (for formatting code, to be more precise)
- The library is not actually distributed with the product
- Users don't install the library when they install the app

Comment (Mblanchetdeverly) [reply public.yml:33]: @jonapich then the MPL is fine for your specific use case. As mentionned before, I cannot approve it at large so we will not add it to the list of approved licenses but you can proceed in your case and ignore the alert with a comment "approved by Legal based on my specific use case".

**Pull request closed without merging**
